DigitalLearningSolutions.Web/Controllers/TrackingSystem/Centre/Dashboard/RankingController.cs

```diff
@@ -12,20 +14,46 @@
     [Route("/TrackingSystem/Centre/Ranking")]
     public class RankingController : Controller
     {
+        private static readonly (int, string) Week = (7, "Week");
+        private static readonly (int, string) Fortnight = (14, "Fortnight");
+        private static readonly (int, string) Month = (30, "Month");
+        private static readonly (int, string) Quarter = (91, "Quarter");
+        private static readonly (int, string) Year = (365, "Year");
+
+        private static readonly IEnumerable<(int, string)> Periods = new[]
+        {
+            Week,
+            Fortnight,
+            Month,
+            Quarter,
+            Year
+        };
+
         private readonly ICentresService centresService;
+        private readonly IRegionDataService regionDataService;
 
-        public RankingController(ICentresService centresService)
+        public RankingController(
+            ICentresService centresService,
+            IRegionDataService regionDataService
+        )
         {
             this.centresService = centresService;
+            this.regionDataService = regionDataService;
         }
 
-        public IActionResult Index()
+        public IActionResult Index(int? regionId = null, int? period = null)
         {
+            period ??= Fortnight.Item1;

```

Comment:
Why not use this as the default value? Does the compiler not like invoking the static quantity?

Reply:
Default value of the parameter must be a compile time constant.

Reply:
That's what I figured. This is better than a magic number then.